lib/push.js

```diff
@@ -35,25 +44,31 @@ module.exports = function (log, db) {
             if (device.pushCallback) {
               // send the push notification
               log.increment(LOG_UPDATE_INCREMENT_SEND)
-              request.post(device.pushCallback, function (err){
+              request.post(device.pushCallback, function (err, response) {
                 if (err) {
-                  log.error({
-                    op: LOG_OP_NOTIFY_UPDATE,
-                    deviceId: deviceId,
-                    err: err
-                  })
-                  log.increment(LOG_UPDATE_INCREMENT_FAIL)
+                  // any 400 level error from the push servers means
+                  // the push settings need to be reset.
+                  // the clients will check this and re-register push endpoints
```

Comment:
The spec suggests at least one case where this is not accurate - according to [1] a `429 Too Many Requests` response indicates that we're being throttled by the push service, but the underlying subscription is still usable.

Unfortunately the spec is very vague [2] about the precise set of 4XX codes that might indicate the subscription has died.  We could either check against a set of known matches (404 and 410 at the very least) or against a set of known non-matches (429 and 413 are mentioned in the webpush spec) but I don't think we can treat all 4XX errors the same.

[1] https://tools.ietf.org/html/draft-ietf-webpush-protocol-02#section-6
[2] https://tools.ietf.org/html/draft-ietf-webpush-protocol-02#section-8.3